File: Programmers/Summer_winter_coding/예산/budget.py
import sys

def solution(d, budget):
    d.sort()
    answer=0
    for money in d:
        if budget-money>=0:
            budget-=money
            answer+=1
        else:
            break
    return answer

d=list(map(int,input().split()))
budget=int(input())
print(solution(d,budget))

File: Programmers/Summer_winter_coding/예산/budget.js
function solution(d, budget) {
	d.sort((a, b) => {
		return a - b;
	});
	let cnt = 0;
	while (budget > 0) {
		if (budget - d[0] >= 0) {
			budget -= d[0];
			d.shift();
			cnt++;
		} else {
			break;
		}
	}
	return cnt;
}

let d = [1, 3, 2, 5, 4];

let budget = 9;
console.log(solution(d, budget));
